jbpm-console-ng-case-mgmt/jbpm-console-ng-case-mgmt-client/src/main/java/org/jbpm/console/ng/cm/client/comments/CaseCommentsPresenter.java

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.console.ng.cm.client.comments;
+
+import java.util.Date;
+import java.util.List;
+import javax.enterprise.context.Dependent;
+import javax.inject.Inject;
+
+import org.jboss.errai.common.client.api.RemoteCallback;
+import org.jboss.errai.security.shared.api.identity.User;
+import org.jboss.errai.ui.client.local.spi.TranslationService;
+import org.jbpm.console.ng.cm.client.AbstractCaseInstancePresenter;
+import org.jbpm.console.ng.cm.model.CaseCommentSummary;
+import org.jbpm.console.ng.cm.model.CaseInstanceSummary;
+import org.uberfire.client.annotations.WorkbenchPartTitle;
+import org.uberfire.client.annotations.WorkbenchPartView;
+import org.uberfire.client.annotations.WorkbenchScreen;
+import org.uberfire.client.mvp.UberView;
+import org.uberfire.ext.widgets.common.client.callbacks.DefaultErrorCallback;
+import org.uberfire.mvp.Command;
+
+import static org.jbpm.console.ng.cm.client.resources.i18n.Constants.*;
+
+@Dependent
+@WorkbenchScreen(identifier = CaseCommentsPresenter.SCREEN_ID)
+public class CaseCommentsPresenter extends AbstractCaseInstancePresenter {
+
+    public static final String SCREEN_ID = "Case Comments";
+
+    @Inject
+    private CaseCommentsView caseCommentView;
+
+    @Inject
+    private TranslationService translationService;
+
+    @Inject
+    User identity;
+
+    private String currentUpdatedCommentId = "";
+
+    @WorkbenchPartView
+    public UberView<CaseCommentsPresenter> getView() {
+        return caseCommentView;
+    }
+
+    @WorkbenchPartTitle
+    public String getTittle() {
+        return translationService.format(CASE_COMMENTS);
+    }
+
+    @Override
+    protected void clearCaseInstance() {
+        caseCommentView.removeAllComments();
+    }
+
+    @Override
+    protected void loadCaseInstance(final CaseInstanceSummary cis) {
+        if (caseId == null) {
+            return;
+        }
+        refreshComments();
+    }
+
+    public void refreshComments() {
+        clearCaseInstance();
+        caseService.call(
+                new RemoteCallback<List<CaseCommentSummary>>() {
```

Comment:
Overall look at using more functional styling, that has performance benefits as well as making the code easier to read/understand.